File: tests/test_duplicated.py
from hetgpy.utils import duplicated
import numpy as np
from rpy2.robjects import r

def test_duplicates():
    # from help(duplicated) in R
    r("x <- c(9:20, 1:5, 3:7, 0:8)")
    x = np.array(r("x"))
    test1 = (duplicated(x) == np.array(r("duplicated(x)")).astype(bool)).all()
    test2 = (duplicated(x,fromLast = True) == np.array(r("duplicated(x,fromLast = T)")).astype(bool)).all()
    assert test1
    assert test2

    # 2d example
    r('X <- matrix(c(c(1,2,3),c(1,2,3),c(4,5,6)),nrow=3,byrow=T)')
    X = np.array(r('X'))
    test3 = (duplicated(X) == np.array(r('duplicated(X)')).astype(bool)).all()
    test4 = (duplicated(X,fromLast=True) == np.array(r('duplicated(X,fromLast=T)')).astype(bool)).all()
    assert test3
    assert test4

File: tests/R/test_duplicated.R
library(yaml)
# 1D example
x <- c(9:20, 1:5, 3:7, 0:8)

D1 = duplicated(x)
D1_L = duplicated(x,fromLast = T)
#2D example

X <- matrix(c(c(1,2,3),c(1,2,3),c(4,5,6)),nrow=3,byrow=T)

D2 = duplicated(X)
D2_L = duplicated(X,fromLast=T)

out  = list(
  x = x,
  X = X,
  D1 = D1,
  D1_L = D1_L,
  D2 = D2,
  D2_L = D2_L
)

cat(as.yaml(out),file = 'results/test_duplicated.yaml')